lib/core/utils/id_helper.dart

```diff
@@ -1,12 +1,23 @@
 // import 'package:device_uuid/device_uuid.dart';
+import 'package:device_uuid/device_uuid.dart';
 import 'package:uuid/uuid.dart';
+import 'package:crypto/crypto.dart';
+import 'dart:convert';
+
+/// Returns a SHA256 hash of the result.
+String hashConcatenatedStrings(String id) {
+  final bytes = utf8.encode(id);
+  final digest = sha256.convert(bytes);
+  return digest.toString();
+}
 
 /// Generates a unique ID by combining the device UUID and a random UUID.
 /// If device UUID is unavailable, returns only the random UUID.
 Future<String> generateDeviceScopedId() async {
-  // final deviceId = await DeviceUuid().getUUID();
+  final deviceId = await DeviceUuid().getUUID();
   final randomId = const Uuid().v4();
 
-  return randomId;
-  // return deviceId != null ? '$deviceId-$randomId' : randomId;
+  // return randomId;
+  final random = deviceId != null ? '$deviceId-$randomId' : randomId;
+  return hashConcatenatedStrings(random);
```

Comment:
```suggestion
  return '$deviceId:$randomId'
```

I know we made this decision to hash together. But After thinking about it the goal here is uniqueness and not privacy. So hashing is not necessary.

The hashing we also use is irreversible so, we can use ID's to scope back to devices if we ever wanted to.


So please let's simplify it and use use `DIVICDE_ID:RANDOM_ID`

@kofimokome this makes the backend validation logic also very easy, and should be documented in https://github.com/trakli/webservice/pull/81